iothub/service/samples/getting started/ServiceClientSample/ServiceClientSample.cs

```diff
@@ -59,40 +58,42 @@ private async Task ReceiveMessageFeedbacksAsync(CancellationToken token)
             // It is important to note that receiver only gets feedback messages when the device is actively running and acting on messages.
             _logger.LogInformation("Starting to listen to feedback messages");
 
-            var feedbackReceiver = _serviceClient.GetFeedbackReceiver();
-
-            while (!token.IsCancellationRequested)
+            AcknowledgementType OnC2dMessageAck(FeedbackBatch feedbackMessages)
             {
+                AcknowledgementType ackType = AcknowledgementType.Abandon;
+
                 try
                 {
-                    FeedbackBatch feedbackMessages = await feedbackReceiver.ReceiveAsync(token);
-                    if (feedbackMessages != null)
+                    _logger.LogInformation("New Feedback received:");
+                    _logger.LogInformation($"\tEnqueue Time: {feedbackMessages.EnqueuedOnUtc}");
+                    _logger.LogInformation($"\tNumber of messages in the batch: {feedbackMessages.Records.Count()}");
+                    foreach (FeedbackRecord feedbackRecord in feedbackMessages.Records)
                     {
-                        _logger.LogInformation("New Feedback received:");
-                        _logger.LogInformation($"\tEnqueue Time: {feedbackMessages.EnqueuedTime}");
-                        _logger.LogInformation($"\tNumber of messages in the batch: {feedbackMessages.Records.Count()}");
-                        foreach (FeedbackRecord feedbackRecord in feedbackMessages.Records)
-                        {
-                            _logger.LogInformation($"\tDevice {feedbackRecord.DeviceId} acted on message: {feedbackRecord.OriginalMessageId} with status: {feedbackRecord.StatusCode}");
-                        }
-
-                        await feedbackReceiver.CompleteAsync(feedbackMessages, token);
+                        _logger.LogInformation($"\tDevice {feedbackRecord.DeviceId} acted on message: {feedbackRecord.OriginalMessageId} with status: {feedbackRecord.StatusCode}");
                     }
 
-                    await Task.Delay(s_sleepDuration, token);
-                }
-                catch (Exception e) when (ExceptionHelper.IsNetwork(e))
-                {
-                    _logger.LogError($"Transient Exception occurred; will retry: {e}");
-
+                    Task.Delay(s_sleepDuration, token);
```

Comment:
Why sleep?

Reply:
I think that was when we were making explicit calls rather than getting C2D messages in a callback, so definitely no longer needed.